elf_reader.go

```diff
@@ -466,19 +466,35 @@ func mapSpecFromBTF(spec *btf.Spec, name string) (*MapSpec, error) {
 		return nil, fmt.Errorf("can't get BTF: %w", err)
 	}
 
+	var btfKeyType, btfValueType btf.Type = (*btf.Void)(nil), (*btf.Void)(nil)
+	var keySize, valueSize uint32
 	keyType := btf.MapKey(btfMap)
-	size, err := btf.Sizeof(keyType)
-	if err != nil {
-		return nil, fmt.Errorf("can't get size of BTF key: %w", err)
+	if pk, isPtr := keyType.(*btf.Pointer); !isPtr {
```

Comment:
`btf.Spec.Map` should unwrap the pointer for Key and Value. Then you don't have to change the code here, and can also avoid having to extend MapSpec.

Reply:
Fixed.